specs/public_inputs.md

```diff
@@ -0,0 +1,263 @@
+# Public Inputs
+
+The following list contains all the data used by an Ethereum client to
+calculate a block and to define this calculated block.
+
+## Block
+
+- Hash: 256 bits
+- Block Fields used in RLP:
+    - ParentHash: 256 bits
+    - UncleHash: 256 bits
+    - Coinbase: 160 bits
+    - Root: 256 bits (State Trie Root)
+    - TxHash: 256 bits (Txs Trie Root)
+    - ReceiptHash: 256 bits (Receipts Trie Root)
+    - Bloom: 256 bytes
+    - Difficulty: 256 bits
+    - Number: 64 bits
+    - GasLimit: 64 bits
+    - GasUsed: 64 bits
+    - Time: 64 bits
+    - Extra: 0 bytes
+    - MixDigest: 256 bits
+    - Nonce: 64 bits
+    - BaseFee: 256 bits (BaseFee was added by EIP-1559 and is ignored in legacy headers.)
+
+### Circuits
+
+- Block Hash verifier
+    - All fields
+- EVM Circuit Block Table
+    - Coinbase
+    - GasLimit
+    - Number
+    - Time
+    - Difficulty
+    - BaseFee
+- State Circuit
+    - Block.Root
+
+## Previous Blocks
+
+- block[-1].Root: 256 bits
+- block[-1..-257].Hash: 256 x 256 bits
+
+### Circuits
+
+- EVM Circuit Block Table
+    - block[-1..-257].Hash
+- State Circuit
+    - block[-1].Root
+
+## Globals
+
+- ChainID: 64 bits
+
+### Circuits
+
+- EVM Circuit Block Table
+    - ChainID
+- TxCircuit
+    - ChainID
+
+## Transactions
+
+- Nonce: 64 bits
+- GasPrice: 256 bits
+- Gas: 64 bits
+- CallerAddress: 160 bits
+- CalleeAddress: 160 bits
+- IsCreate: 1 bit
+- Value: 256 bits
+- CallDataLength: 64 bits
+- CallData: CallDataLength bytes
+- TxSignHash: 256 bits
+- Signature
+    - v: 256 bits
+    - r: 256 bits
+    - s: 256 bits
+
+Fields used for RLP for TxSignHash:
+- Nonce
+- GasPrice
+- Gas
+- CalleeAddress
+- Value
+- CallData
+- ChainID
+
+Fields used for RLP for TxHash:
```

Comment:
is it only for `LegacyTx` type ?  since `AccessListTx` type may have access list property

Reply:
You're correct!  I've added a note saying that for now we only consider legacy transactions in https://github.com/privacy-scaling-explorations/zkevm-specs/commit/849a90149a20adb3506108a5bf1a037a13382e7a

Reply:
> Good catch! I also wonder that @ed255 I guess depends on the hard fork we want to provide support for at the begging. `London`? `Berlin`?

The idea is to start with support for legacy transaction only to make things a bit simpler; and then in the future figure out a way to support versioned transactions.